MatomoTracker/UserDefaultsBackedQueue.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+public final class UserDefaultsBackedQueue: NSObject, Queue {
+    let userDefaults: UserDefaults
+    
+    init(suiteName: String?) {
+        userDefaults = UserDefaults(suiteName: suiteName)!
+    }
```

Comment:
What do you think about passing in an instance of `UserDefaults` directly?
```suggestion
    init(userDefaults: UserDefaults) {
        userDefaults = userDefaults
    }
```